liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,383 @@
+const {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+} = require('forta-agent');
+const BigNumber = require('bignumber.js');
+const config = require('../agent-config.json');
+const {
+  getAbi, callCompoundAPI, buildJsonRequest,
+} = require('./utils');
+
+// Stores information about each account
+const initializeData = {};
+
+// Global functions
+function createAlert(
+  developerAbbreviation,
+  protocolName,
+  protocolAbbreviation,
+  type,
+  severity,
+  borrowerAddress,
+  liquidationAmount,
+  shortfallAmount,
+  healthFactor,
+) {
+  return Finding.fromObject({
+    name: `${protocolName} Liquidation Threshold Alert`,
+    description: `The address ${borrowerAddress} has dropped below the liquidation threshold. `
+      + `The account may be liquidated for: $${liquidationAmount} USD`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-LIQUIDATION-THRESHOLD`,
+    type: FindingType[type],
+    severity: FindingSeverity[severity],
+    protocol: protocolName,
+    metadata: {
+      borrowerAddress,
+      liquidationAmount,
+      shortfallAmount,
+      healthFactor,
+    },
+  });
+}
+
+async function verifyToken(data, tokenAddressImport) {
+  const { tokens } = data;
+  const tokenAddress = tokenAddressImport.toLowerCase();
+  if (tokens[tokenAddress] === undefined) {
+    tokens[tokenAddress] = {};
+    tokens[tokenAddress].contract = new ethers.Contract(
+      tokenAddress, data.cTokenABI, data.provider,
+    );
+    const cContract = tokens[tokenAddress].contract;
+    tokens[tokenAddress].symbol = await cContract.symbol();
+
+    // cETH does not have an underlying contract, so link it to wETH instead
+    if (tokenAddress === '0x4ddc2d193948926d02f9b1fe9e1daa0718270ed5') {
+      tokens[tokenAddress].underlying = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
+    } else {
+      tokens[tokenAddress].underlying = await cContract.underlying();
+    }
+    const exchangeRate = await cContract.exchangeRateStored();
+    tokens[tokenAddress].cTokenDecimals = BigNumber(
```

Comment:
Although JavaScript will run this, the constructor for BigNumber should use the `new` keyword:
```suggestion
    tokens[tokenAddress].cTokenDecimals = new BigNumber(
```

This should be done anywhere the BigNumber constructor is used.

Reply:
Corrected on commit 68d8ca5da630a046bf35cbf095720fa39beccb37